cmd/algorand-indexer/daemon.go

```diff
@@ -174,8 +178,15 @@ func (bih *blockImporterHandler) HandleBlock(block *types.EncodedBlockCert) {
 		logger.Errorf("received block %d when expecting %d", block.Block.Round, bih.round+1)
 	}
 	_, err := bih.imp.ImportDecodedBlock(block)
-	maybeFail(err, "ImportDecodedBlock %d: %v", block.Block.Round, err)
-	importer.UpdateAccounting(bih.db, bih.cache, uint64(block.Block.Round), genesisJSONPath, logger)
+	maybeFail(err, "ImportDecodedBlock %d", block.Block.Round)
+	maxRoundAccounted, err := bih.db.GetMaxRoundAccounted()
+	maybeFail(err, "failed to get max round accounted.")
+	// During normal operation StartRound and MaxRound will be the same round.
+	filter := idb.UpdateFilter{
+		StartRound: maxRoundAccounted + 1,
```

Comment:
I don't think this is consistent for the initial case. With no prior state, maxRoundAccounted will be 0, but yieldTxns returns > round, so round needs to be -1, so StartRound needs to be 0.

Reply:
I liked the name `StartRound` so I flipped the off-by-one to `prevRound` in UpdateAccounting:
```
txns := db.YieldTxns(context.Background(), int64(filter.StartRound)-1)
```

Reply:
I think this could miss txns in round=1 (or round=0) when rebuilding after reset.

Reply:
it took me a while but I finally see it... 0 vs none strikes again...

Reply:
Fixed by having GetMaxRoundAccounted return ErrorNotInitialized.